modules/indexer/code/bleve.go

```diff
@@ -86,16 +91,23 @@ func openIndexer(path string, latestVersion int) (bleve.Index, error) {
 
 // RepoIndexerData data stored in the repo indexer
 type RepoIndexerData struct {
-	RepoID  int64
-	Content string
+	RepoID    int64
+	CommitID  string
+	Content   string
+	Language  string
+	UpdatedAt time.Time
 }
 
 // Type returns the document type, for bleve's mapping.Classifier interface.
 func (d *RepoIndexerData) Type() string {
 	return repoIndexerDocType
 }
 
-func addUpdate(update fileUpdate, repo *models.Repository, batch rupture.FlushingBatch) error {
+func addUpdate(commitSha string, update fileUpdate, repo *models.Repository, batch rupture.FlushingBatch) error {
+	// Ignore vendored files in code search
```

Comment:
I would like user to define them on `.gitea/indexer` file on this repository. It should be a .gitignore like file format and I think template repository has implemented it.

All matched files on this file will be ignored by code/language detection indexer.

Reply:
That would be nice. Perhaps path patterns are also useful? For example, Gitea's code is mostly go, but there's a lot of go files inside `./vendor` that one might not want indexed.

Reply:
@guillep2k Yes. If that file exist, it will override configuration on ini which feature you implemented before.

Reply:
This should be probably left to improve in other PR